platform/admin/core/admin-core-insecuredefaults/src/test/java/org/codice/ddf/admin/insecure/defaults/service/UserPropertiesDeletionSchedulerTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.insecure.defaults.service;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertNull;
+import static junit.framework.TestCase.assertTrue;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class UserPropertiesDeletionSchedulerTest {
+
+  private static final Path PATH = UserPropertiesDeletionScheduler.USERS_PROPERTIES_FILE_PATH;
+  private static final Path TEMP_FILE_PATH =
+      UserPropertiesDeletionScheduler.TEMP_TIMESTAMP_FILE_PATH;
```

Comment:
❗️  Instead of doing this, you should create a temporary directory using the junit rule and work with arbitrary directories / location within a temporary root. It ensures things get cleaned up on the host correctly. It looks like this:
```
@Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
```
It also means you need not expose your constants for testing.

Reply:
I can use that but I still need the temp file to be in a specific place so I need to specify a path for `TemporaryFolder`